pkg/executor/infoschema_reader.go

```diff
@@ -3631,6 +3633,43 @@ func (e *memtableRetriever) setDataForClusterIndexUsage(ctx sessionctx.Context,
 	return nil
 }
 
+func (e *memtableRetriever) setDataForKeyspaceMeta(sctx sessionctx.Context) (err error) {
+	meta := sctx.GetStore().GetCodec().GetKeyspaceMeta()
```

Comment:
Can it push down the predicate? If cluster has a large number of keyspace, has it problem?